File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.java
class Solution {
    public String solution(String s, String skip, int index) {
        StringBuilder answer = new StringBuilder();
        
        for(char c : s.toCharArray()){
            char check = c;
            int idx = 0 ;
            
            while(idx < index){
                check=check=='z'?'a':(char)(check+1);
                if(!skip.contains(String.valueOf(check))){
                    idx +=1;
                }
            }
            answer.append(check);
            
        }
        return answer.toString();
    }
}

File: C#/프로그래머스/1/155652. 둘만의 암호/둘만의 암호.cs
using System;

public class Solution {
    public string solution(string s, string skip, int index) {
string answer = "";

            for(int i = 0; i < s.Length; i++)
            {
                int num = 1;
                char c = s[i];
                while (num <= index)
                {
                    ++c;
                    if (c - '0' > 74) c = 'a';
                    if (skip.Contains(c)) continue;

                    num++;
                }
                answer += c;
            }

            return answer;
    }
}